extensions/amp-pan-zoom/0.1/amp-pan-zoom.js

```diff
@@ -644,39 +662,38 @@ export class AmpPanZoom extends AMP.BaseElement {
   }
 
   /**
-   * Performs a one-step pinch zoom action.
-   * @param {number} centerClientX
-   * @param {number} centerClientY
-   * @param {number} deltaX
-   * @param {number} deltaY
-   *  @param {number} dir
-   * @private
+   * @param {number} clientX
+   * @param {number} clientY
    */
-  onPinchZoom_(centerClientX, centerClientY, deltaX, deltaY, dir) {
-    this.zoomToPoint_(centerClientX, centerClientY, deltaX, deltaY, dir);
+  onDoubletapZoom_(clientX, clientY) {
+    const newScale = this.scale_ == 1 ? this.maxScale_ : this.minScale_;
+    const dx = (this.elementBox_.width / 2) + this.getOffsetX_(clientX);
+    const dy = (this.elementBox_.height / 2) + this.getOffsetY_(clientY);
+    return this.onZoom_(newScale, dx, dy, /*animate*/ true);
   }
 
   /**
-   * Given center position, zoom delta, and zoom position, computes
-   * and updates a zoom action on the content.
+   * Performs a one-step pinch zoom action.
    * @param {number} centerClientX
    * @param {number} centerClientY
    * @param {number} deltaX
    * @param {number} deltaY
-   * @param {number} dir
+   *  @param {number} dir
    * @private
    */
-  zoomToPoint_(centerClientX, centerClientY, deltaX, deltaY, dir) {
+  onPinchZoom_(centerClientX, centerClientY, deltaX, deltaY, dir) {
     if (dir == 0) {
       return;
     }
+    const {width, height} = this.elementBox_;
     const dist = Math.sqrt((deltaX * deltaX) + (deltaY * deltaY));
-    const newScale = this.startScale_ * (1 + (dir * dist / 100));
-    const deltaCenterX = (this.elementBox_.width / 2) - centerClientX;
-    const deltaCenterY = (this.elementBox_.height / 2) - centerClientY;
-    deltaX = Math.min(deltaCenterX, deltaCenterX * (dist / 100));
-    deltaY = Math.min(deltaCenterY, deltaCenterY * (dist / 100));
-    this.onZoom_(newScale, deltaX, deltaY, /*animate*/ false);
+    const pinchFactor = dist / 100;
+    const newScale = this.startScale_ * (1 + (dir * pinchFactor));
+    const deltaCenterX = width / 2 + this.getOffsetX_(centerClientX);
+    const deltaCenterY = height / 2 + this.getOffsetY_(centerClientY);
+    const dx = pinchFactor > 1 ? deltaCenterX : deltaCenterX * pinchFactor;
```

Comment:
Might be a bit nicer to remove the conditional here (within your code). For example:

```javascript
const dx = Math.min(pinchFactor, 1) * deltaCenterX;
```

It feels a bit more clear to me that pinchFactor is capped to 1. What isn't clear is why.